backend/src/poll/poll.controller.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  Patch,
+  Post,
+  Res,
+} from '@nestjs/common';
+import {
+  ApiCreatedResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import Poll from '../models/poll';
+import { CreatePollDto } from './dto/create-poll.dto';
+import { UpdatePollDto } from './dto/update-poll.dto';
+import { PollService } from './poll.service';
+import { Response } from 'express';
+
+@ApiTags('polls')
+@Controller('polls')
+export class PollController {
+  constructor(private readonly pollService: PollService) {}
+
+  @Post()
+  @ApiCreatedResponse({ type: Poll, description: 'The poll has been created' })
+  async create(@Body() createPollDto: CreatePollDto): Promise<Poll> {
+    return await this.pollService.create(createPollDto);
+  }
+
+  @Get()
+  @ApiOkResponse({
+    type: [Poll],
+    description: 'Returns all polls of the system',
+  })
+  async findAll(): Promise<Poll[]> {
+    return await this.pollService.findAll();
+  }
+
+  @Get(':pollId')
+  @ApiOkResponse({ type: Poll, description: 'Returns the poll with this id' })
+  @ApiNotFoundResponse({
+    type: null,
+    description: 'No poll with this id has been found',
+  })
+  async findOne(
+    @Param('pollId') pollId: number,
+    @Res({ passthrough: true }) res: Response,
+  ): Promise<Poll | void> {
+    const poll = await this.pollService.findOne(pollId);
+    if (!poll) {
+      res.status(HttpStatus.NOT_FOUND);
+      return;
+    }
+
+    return poll;
+  }
+
+  @Get('byCode/:code')
```

Comment:
This endpoint is very similar to `findOne`. Maybe it is possible to merge the two? Will they have different semantics int the future?

Reply:
The idea behind the `/polls/pollId` route was so that the frontend can always find the current poll by the id. `/polls/byCode/code` should only be used once to find the poll initially.